lib/flight_job/outputs/info_job.rb

```diff
@@ -28,107 +28,99 @@
 require 'output_mode'
 
 module FlightJob
-  module Outputs::InfoJob
-    extend OutputMode::TLDR::Show
-
-    TEMPLATE = <<~ERB
-      <% verbose = output.context[:verbose] -%>
-      <% each(:default) do |value, padding:, field:| -%>
-      <%
-          # Apply the colours
-          value = pastel.green value
-          field = pastel.blue.bold field
-      -%>
-      <%= padding -%><%= pastel.blue.bold field -%><%= pastel.bold ':' -%> <%= value %>
-      <% end -%>
-      <%
-        submit_outputs = output.procs.select { |proc| proc.config[:section] == :submit }
-        if verbose || output.context[:submit]
-      -%>
-
-      <%= pastel.blue.bold "Submit Stdout" -%><%= pastel.bold ':' %>
-      <%= pastel.green submit_outputs.first.call(model) %>
-      <%= pastel.blue.bold "Submit Stderr" -%><%= pastel.bold ':' %>
-      <%= pastel.green submit_outputs.last.call(model) %>
-      <% end -%>
-    ERB
-
-    register_attribute(header: 'ID') { |j| j.id }
-    register_attribute(header: 'Script ID') { |j| j.script_id }
-    register_attribute(header: 'Scheduler ID') { |j| j.scheduler_id }
-    register_attribute(header: 'State') { |j| j.state }
-
-    # Show a boolean in the "simplified" output, and the exit code in the verbose
-    # NOTE: There is a rendering issue of integers into the TSV output. Needs investigation
-    register_attribute(header: 'Submitted', verbose: false) { |j| j.submit_status == 0 }
-    register_attribute(header: 'Submit Status', verbose: true) { |j| j.submit_status.to_s }
-
-    register_attribute(header: 'Submitted at') do |job, verbose:|
-      if verbose
-        job.created_at
-      else
-        DateTime.rfc3339(job.created_at).strftime('%d/%m/%y %H:%M')
-      end
+  class Outputs::InfoJob < OutputMode::Formatters::Show
+    def initialize(job, submit: true, **opts)
+      @submit = submit
+      super(job, **opts)
+    end
+
+    def submit?
+      @submit ? true : false
     end
 
-    start_header = ->(job, verbose:) do
-      job.actual_start_time || verbose ? 'Started at' : 'Estimated Start'
+    def format_row(field, value, padding)
+      f = pastel.blue.bold field
+      s = pastel.bold ':'
+      v = pastel.green value
+      "#{padding}#{f}#{s} #{v}"
     end
-    register_attribute(header: start_header) do |job, verbose:|
-      if job.actual_start_time || verbose
-        Outputs.format_time(job.actual_start_time, verbose)
+
+    alias_method :job, :object
+
+    constructor do
+      template(<<~ERB) if interactive?
+        <% each(:default) do |value, padding:, field:| -%>
+        <%= format_row(field, value, padding) %>
+        <% end -%>
+        <%
+          submit_outputs = callables.select { |proc| proc.config[:section] == :submit }
+          if verbose? || submit?
+        -%>
```

Comment:
What's the reason for special handling of the stdout and stderr in the template?  Wouldn't it be easier to branch in the `constructor` block?

```ruby
constructor do
  # ....
  if verbose? || submit?
    register(header: 'Submit stdout') { job.submit_stdout }
  end
end
```

Reply:
The formatting is slightly different. The other keys are rendered inline: `ID: foo-job`

Where the `STDOUT`/`STDERR` are offline:

```
STDOUT:
<content>
```

---

There needs to be special casing regardless, so I embedded it into the template

Reply:
Cleaned the template up a bit:

9419cb7ac23e717c30620de954207dbd5df3e954